### PR TITLE
Feature/create dapp data source

### PR DESCRIPTION
**Pull request by kaliumxyz**
Small fixes and adds google sheets data source.

### CODE REVIEW

File: apps/api-consumer-producer/src/app/project/project.service.ts [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+                let project = await _project.findOne({ where: { name: row.name } })
+                delete row.category
+
+                try {
+                    if (project === void 0) {
+                        row.categoryId = category.id
+                        await _project.save(row)
+                    } else {
+                        delete row.externalLinks
```
Comment (stupidawesome) [inline apps/api-consumer-producer/src/app/project/project.service.ts:258]: Allow links to be updated if they exist
```suggestion
row.externalLinks = project.externalLinks.map((link) => {
    return { ...link, url: row[link.name] }
})
```

File: apps/api/src/app/database/entity/project.entity.ts [hunk 2]
```diff
@@ -40,6 +40,9 @@ export class ProjectEntity {
     @Column({ length: 32 })
     public network: string
 
+    @Column("text", { default: "a future project" })
```
Comment (stupidawesome) [inline apps/api/src/app/database/entity/project.entity.ts:43]: Empty string default is fine

**Review by stupidawesome — COMMENTED** [review]
Looking good, see comments:

- Run code formatter please.
- Resolve package-lock conflict. Use correct node version when running npm install.
- Create a migration for the project entity.

**Comment by kaliumxyz** [comment]
> Run code formatter please.

I did https://asciinema.org/a/wWynOvTmLsVNE7zU2zwvoWf4v

> Resolve package-lock conflict. Use correct node version when running npm install.

I'm on v10.15.3 for generating the lock file, NPM is cursed.
https://asciinema.org/a/cEWLxo5g9X4qKHA2JrhJHpFAt

> Create a migration for the project entity.

doing next, let me amend this and change the database behavior before anyone pulls.

File: apps/api/src/app/database/entity/project.entity.ts (hunk 2 above)
Comment (kaliumxyz) [reply apps/api/src/app/database/entity/project.entity.ts:43]: Yea, filler should be handled on the front-end

**Comment by kaliumxyz** [comment]
merged it, will open new PR for sync with google sheets resulting in the db having the same state as the sheet all the time.

**Pull request merged**